src/services/cssFolding.ts

```diff
@@ -79,11 +79,24 @@ export function getFoldingRegions(document: TextDocument): FoldingRangeList {
 					}
 
 					if (startLine !== endLine) {
-						ranges.push({
-							startLine,
-							endLine,
-							type: undefined
-						});
+						if (maxRanges) {
```

Comment:
Of course we can just cut the last regions, but that is not the best user experience.
In HTML and JSON implementatio I filter out the most nested ranges.

Reply:
Extracted to #90.